tests/test_api.py

```
@@ -218,17 +223,58 @@ def test_browse_page_renders_facets(monkeypatch):
+    assert r.status_code == 200
+    assert "Document detail" in r.text
+    assert "research/paper.md" in r.text
+    assert "/tmp/private/path" not in r.text
+    assert "First chunk with enough text to preview." in r.text
+    assert "chunk-2" in r.text
+
+
+def test_browse_document_detail_404(monkeypatch):
+    from alcove.query import api as api_mod
+
+    monkeypatch.setattr(api_mod, "browse_document_detail", lambda document_id: None)
```

Ruff (0.15.12)

[warning] Unused lambda argument: `document_id`

(ARG005)